admin_tests/preprints/test_views.py

```diff
@@ -251,3 +257,111 @@ def test_reindex_preprint_share(self, mock_reindex_preprint, preprint, req):
 
         assert mock_reindex_preprint.called
         assert AdminLogEntry.objects.count() == count + 1
+
+@pytest.mark.urls('admin.base.urls')
+class TestPreprintWithdrawalRequests:
+
+    @pytest.fixture()
+    def submitter(self):
+        return AuthUserFactory()
+
+    @pytest.fixture()
+    def admin(self):
+        admin = AuthUserFactory()
+        osf_admin = Group.objects.get(name='osf_admin')
+        admin.groups.add(osf_admin)
+        return admin
+
+    @pytest.fixture()
+    def project(self, submitter):
+        return NodeFactory(creator=submitter)
+
+    @pytest.fixture()
+    def preprint(self, project):
+        return PreprintFactory(project=project)
+
+    @pytest.fixture()
+    def withdrawal_request(self, preprint, submitter):
+        withdrawal_request = PreprintRequestFactory(
+            creator=submitter,
+            target=preprint,
+            request_type=RequestTypes.WITHDRAWAL.value,
+            machine_state=DefaultStates.INITIAL.value,
+        )
+        withdrawal_request.run_submit(submitter)
+        return withdrawal_request
+
+    def test_can_approve_withdrawal_request(self, withdrawal_request, submitter, preprint, admin):
+        assert withdrawal_request.machine_state == DefaultStates.PENDING.value
+
+        request = RequestFactory().post(reverse('preprints:approve-withdrawal', kwargs={'guid': preprint._id}))
+        request.user = admin
+
+        response = views.PreprintApproveWithdrawalRequest.as_view()(request, guid=preprint._id)
+        assert response.status_code == 302
+
+        withdrawal_request.refresh_from_db()
+        withdrawal_request.target.refresh_from_db()
+        assert withdrawal_request.machine_state == DefaultStates.ACCEPTED.value
+        assert withdrawal_request.comment == withdrawal_request.target.withdrawal_justification
```

Comment:
So, I've confirmed that this process does work now, but this test isn't actually ensuring that it doesn't change on approval.

To do so, you'd want to set something like `original_justification = withdrawal_request.comment` around the start of this test, then check the `target.withdrawal_justification` against that var after the approval view is hit.